v2/generator/package_registry.go

```diff
@@ -0,0 +1,123 @@
+package generator
+
+import (
+	"fmt"
+	"sort"
+
+	"go.mercari.io/yo/v2/models"
+)
+
+var defaultTypeModulePackages = []*models.Package{
+	{Path: "context"},
+	{Path: "fmt"},
+	{Path: "strings"},
+	{Path: "time"},
+
+	{Path: "cloud.google.com/go/spanner"},
+	{Path: "google.golang.org/api/iterator"},
+	{Path: "google.golang.org/grpc/codes"},
+}
+
+var defaultGlobalModulePackages = []*models.Package{
+	{Path: "context"},
+	{Path: "errors"},
+	{Path: "fmt"},
+	{Path: "strconv"},
+
+	{Path: "cloud.google.com/go/spanner"},
+	{Path: "github.com/googleapis/gax-go/v2/apierror"},
+	{Path: "google.golang.org/grpc/codes"},
+	{Path: "google.golang.org/grpc/status"},
+}
+
+// NewTypePackageRegistry creates a new instance of PackageRegistry for a specific type module
+func NewTypePackageRegistry(t *models.Type) (*PackageRegistry, error) {
+	r, err := newPackageRegistry(defaultTypeModulePackages)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, f := range t.Fields {
+		if f.Package == nil {
+			continue
+		}
+
+		if err := r.Register(f.Package); err != nil {
```

Comment:
In this PR, it's enough to register the packages in advance. However, if we consider the custom module, the packages are possibly not used by the generated code at all. So it needs to be registered when the package is really used. (Even if there is unused package in import, they wil be removed automatically by goimports though)